### PR TITLE
refactor: remove not null annotations

### PR DESCRIPTION
**Pull request by ChristianMurphy**
##### Checklist

- [x] the [contributor license agreement][] is signed
- [x] commit message follows [commit guidelines][]
- [ ] tests are included
- [ ] screenshots are included showing significant UI changes
- [ ] documentation is changed or added

##### Description of change

https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/no-non-null-assertion.md

[contributor license agreement]: https://www.apereo.org/node/676
[commit guidelines]: https://chris.beams.io/posts/git-commit/

### CODE REVIEW

**Comment by ChristianMurphy** [comment]
Thanks for the feedback @edalex-ian! :bow: 
the code has been updated to ignore the lint errors in *cloudcontrol/index.ts* and resolve the errors elsewhere.

File: Source/Plugins/Core/com.equella.core/js/tsrc/settings/Search/searchfilter/SearchFilterSettingsPage.tsx [hunk 1]
```diff
@@ -250,7 +250,12 @@ const SearchFilterPage = ({ updateTemplate }: TemplateUpdateProps) => {
         (): Promise<any> =>
           // Filters stored in 'deletedMimeTypeFilters' always have an ID.
           deletedMimeTypeFilters.length
-            ? batchDelete(deletedMimeTypeFilters.map((filter) => filter.id!))
+            ? batchDelete(
+                deletedMimeTypeFilters.map(({ id }) => {
+                  if (!id) throw TypeError("id is undefined");
+                  return id;
+                })
```
Comment (edalex-ian) [inline Source/Plugins/Core/com.equella.core/js/tsrc/settings/Search/searchfilter/SearchFilterSettingsPage.tsx:257]: With this being duplicated, it feels like `batchDelete` should simply be taking a type which is a collection of objects with an `id` field which it then does the above transform on.

Any chance you could include something like that?

Comment (ChristianMurphy) [reply Source/Plugins/Core/com.equella.core/js/tsrc/settings/Search/searchfilter/SearchFilterSettingsPage.tsx]: Something to note, the two batchDelete functions are named the same but are different.
They also input different types, you can take a look at what shared functionality would look like in ce8773b.

It is possible to share the functionality.
It's moderately verbose to do so.

Comment (ChristianMurphy) [reply Source/Plugins/Core/com.equella.core/js/tsrc/settings/Search/searchfilter/SearchFilterSettingsPage.tsx]: Thoughts @edalex-ian? :thought_balloon:

Comment (edalex-ian) [reply Source/Plugins/Core/com.equella.core/js/tsrc/settings/Search/searchfilter/SearchFilterSettingsPage.tsx]: Aaaahhhhhh, I see I see I see. Also needs to be moved out into the REST library.

Okay, I like what you've done. :+1:

**Pull request merged**